src/ui/components/filters/sortoptionscomponent.js

```diff
@@ -21,9 +22,16 @@ export default class SortOptionsComponent extends Component {
     this.options[this.selectedOptionIndex].isSelected = true;
     this.hideExcessOptions = this._config.showMore && this.selectedOptionIndex < this._config.showMoreLimit;
     this.showReset = this._config.showReset && this.selectedOptionIndex !== 0;
+
+    /**
+     * This component listens to updates to vertical results, and sets its state to it when
+     * an update occurs.
+     * @type {string}
+     */
+    this.moduleId = StorageKeys.VERTICAL_RESULTS;
   }
 
-  setState (data) {
+  setState (data = {}) {
```

Comment:
Do we need to update the HBS template for this component? If resultsContext is NO_RESULTS, could we just return here instead of calling super.setState? That'd save us having to clutter the HBS with more conditional logic.

Reply:
Tried it, I don't think things work like that because the component's `render()` method is called during `mount()`, and `mount()` is called when `this._state.on('update', () =>`, which needs the setState call. Even then the template still needs to have some conditional logic otherwise everything will render. Basically fake react strikes again.